File: app/src/main/java/com/app/arejaybee/character_sheet/DataObjects/Companion.java
package com.app.arejaybee.character_sheet.DataObjects;

import android.content.Context;

public class Companion extends PlayerCharacter {

    PlayerCharacter m_owner;
    int indexForOwner;

    public Companion(PlayerCharacter owner, int index, String edition, boolean generateSkills) {
        super(edition, generateSkills);
        setOwner(owner);
        indexForOwner = index;
    }

    @Override
    public void saveCharacter(Context con) {
        m_owner.setCompanionAt(indexForOwner, this);
        m_owner.saveCharacter(con);
    }

    public PlayerCharacter getOwner() {
        return m_owner;
    }

    public void setOwner(PlayerCharacter pc) {
        m_owner = pc;
    }

    public boolean hasOwner() {
        return getOwner() != null;
    }

}

File: app/src/main/java/com/app/arejaybee/character_sheet/data_objects/Companion.kt
package com.app.arejaybee.character_sheet.data_objects

import kotlinx.serialization.Serializable

@Serializable
class Companion(var owner: PlayerCharacter, var index: Int) : PlayerCharacter(owner.edition) {
    override fun saveCharacter() {
        owner.let {
            it.companions[index] = this
            it.saveCharacter()
        }
    }
}